### DISCUSSION TITLE
Is there a way to change the default text object `%`, also a second question

### DISCUSSION THREAD

**Discussion by oredaze** [andymass/vim-matchup#231] [Q&A] [answered]
I get why it is `%`, but for me that is not ergonomic at all. I would like to change it to `b` for _block_ or  _brackets_. So for example `ci%` becomes `cib`. Waay easier to type.

Second question: I notice that matchup's commands don't act like vim's native `ci({[` commands in the sense that if no pair is present under the cursor, vim first goes to it and _then_ it executes the command, kinda like first doing `z%` and then `ci%`. I suppose I can do some custom binding, but maybe there is already a way to do it? It seems like a good default functionality.
Maybe this should be a feature request, what do you think?

**Comment by andymass** [comment, marked as answer]
(1) is already supported, but you'll have to specify each customization individually; for instance:

    xmap ib <plug>(matchup-i%)
    omap ib <plug>(matchup-i%)
    xmap ab <plug>(matchup-a%)
    omap ab <plug>(matchup-a%)

There are not that many, perhaps 20 maps at most and depends on your preferences.  We cannot possibly have an option to override all `%` maps, since it would also override `b` itself which is non-sense.

(2) Let me see if this can be supported.  Initially it seems like it would be rarely useful since most of the time you are inside some `{ }` editing code.  But then it wouldn't hurt either.
